### PR TITLE
Use inline for direction functions in Location.hpp

### PR DESCRIPTION
**Pull request by ZehMatt**
Functions that are inside headers are best declared inline. This eliminates the need of [[maybe_unused]] since you are no longer forcing copies of the function on each include, a different library such as libopenrct2ui might not call any of the functions but they are still compiled because of the static keyword and causes unreferenced function warnings by that. This is one of the files where I noticed this and there might be probably more.

### CODE REVIEW

**Review by tupaschoal — COMMENTED** [review]
> Functions that are inside headers are best declared inline.

I mean, at least on clang, using the `inline` keyword used to just bump the threshold that the compiler decided whether to actually inline or not, it was never meant as a guarantee of inlining.

> This eliminates the need of [[maybe_unused]] since you are no longer forcing copies of the function on each include, a different library such as libopenrct2ui might not call any of the functions but they are still compiled because of the static keyword and causes unreferenced function warnings by that. This is one of the files where I noticed this and there might be probably more.

Would it be fine if we just left it as `constexpr`?

**Comment by ZehMatt** [comment]
> > Functions that are inside headers are best declared inline.
> 
> I mean, at least on clang, using the `inline` keyword used to just bump the threshold that the compiler decided whether to actually inline or not, it was never meant as a guarantee of inlining.
I know that it might not inline it but it will also suppress the one definition rule which is the point, unlike static the compiler is not doing anything if its not used.

> Would it be fine if we just left it as `constexpr`?
Nope

**Review by tupaschoal — APPROVED** [review]
LGTM, not sure if you want to wait for a few to leave room for others to review

**Review by Gymnasiast — APPROVED** [review]
It does look a bit cleaner this way.

**Pull request merged**
